File: lesson_1/code/task_1_4_fibonacci_plot.m
% require element count
n = 20;
% 1 st and 2 nd elements are 0 and 1
fib_seq = [1 ,1];
for i = 3: n
fib_seq ( i ) = fib_seq (i -1) + fib_seq (i -2) ;
end

fib_seq_15 = fib_seq(1:15)
plot(fib_seq_15)

ratio = fib_seq_15(2:end)./fib_seq_15(1:end-1);
display(ratio)
plot(ratio, '-s')

File: lesson_1/code/task_1_4_fibonacci_plot.py
import matplotlib.pyplot as plt
import numpy as np

n = 20
fib_seq = [1, 1]
for i in range(2, n):
    fib_seq.append(fib_seq[i-1] + fib_seq[i-2])

fib_seq_15 = np.array(fib_seq[:15])
plt.figure()
plt.plot(fib_seq_15, '-s')
plt.show()

# Calculate ratio (F_n / F_{n-1})
ratio = fib_seq_15[1:] / fib_seq_15[:-1]
print(ratio)

plt.figure()
plt.plot(ratio, '-s')
plt.show()